src/TransientFETools/ODEOperatorInterfaces.jl

```diff
@@ -64,3 +64,125 @@ function jacobian_and_jacobian_t!(J::AbstractMatrix,op::ODEOpFromFEOp,t::Real,uh
+  Ut = get_trial(op.feop)
+  Vt = ∂t(Ut)
+  At = ∂tt(Ut)
+  U = allocate_trial_space(Ut)
+  V = allocate_trial_space(Vt)
+  A = allocate_trial_space(At)
+  fecache = allocate_cache(op.feop)
+  ode_cache = (U,V,A,Ut,Vt,At,fecache)
+  (v,a, ode_cache)
+end
+
+function update_cache!(ode_cache,op::SecondOrderODEOpFromFEOp,t::Real)
+  v,a, (U,V,A,Ut,Vt,At,fecache) = ode_cache
+  U = evaluate!(U,Ut,t)
+  V = evaluate!(V,Vt,t)
+  A = evaluate!(A,At,t)
+  fecache = update_cache!(fecache,op.feop,t)
+  (v,a,(U,V,A,Ut,Vt,At,fecache))
+end
+
+function allocate_residual(op::SecondOrderODEOpFromFEOp,uhF::AbstractVector,ode_cache)
+  U,V,A,Ut,Vt,At,fecache = ode_cache
+  uh = EvaluationFunction(U,uhF)
+  allocate_residual(op.feop,uh,fecache)
+end
+
+function allocate_jacobian(op::SecondOrderODEOpFromFEOp,uhF::AbstractVector,ode_cache)
+  U,V,A,Ut,Vt,At,fecache = ode_cache
+  uh = EvaluationFunction(U,uhF)
+  allocate_jacobian(op.feop,uh,fecache)
+end
+
+function residual!(
+  b::AbstractVector,
+  op::SecondOrderODEOpFromFEOp,
+  t::Real,
+  uhF::AbstractVector,
+  uhtF::AbstractVector,
+  uhttF::AbstractVector,
+  ode_cache)
+  Uh,Uht,Uhtt, = ode_cache
+  uh = EvaluationFunction(Uh,uhF)
+  uht = EvaluationFunction(Uht,uhtF)
+  uhtt = EvaluationFunction(Uhtt,uhttF)
+  residual!(b,op.feop,t,uh,uht,uhtt,ode_cache)
+end
+
+function jacobian!(
+  A::AbstractMatrix,
+  op::SecondOrderODEOpFromFEOp,
+  t::Real,
+  uhF::AbstractVector,
+  uhtF::AbstractVector,
+  uhttF::AbstractVector,
+  ode_cache)
+  Uh,Uht,Uhtt, = ode_cache
+  uh = EvaluationFunction(Uh,uhF)
+  uht = EvaluationFunction(Uht,uhtF)
+  uhtt = EvaluationFunction(Uhtt,uhttF)
+  jacobian!(A,op.feop,t,uh,uht,uhtt,ode_cache)
+end
+
+function jacobian_t!(
+  J::AbstractMatrix,
+  op::SecondOrderODEOpFromFEOp,
+  t::Real,
+  uhF::AbstractVector,
+  uhtF::AbstractVector,
+  uhttF::AbstractVector,
+  dut_u::Real,
+  ode_cache)
+  Uh,Uht,Uhtt, = ode_cache
+  uh = EvaluationFunction(Uh,uhF)
+  uht = EvaluationFunction(Uht,uhtF)
+  uhtt = EvaluationFunction(Uhtt,uhttF)
+  jacobian_t!(J,op.feop,t,uh,uht,uhtt,dut_u,ode_cache)
+end
+
+function jacobian_tt!(
+  J::AbstractMatrix,
+  op::SecondOrderODEOpFromFEOp,
+  t::Real,
+  uhF::AbstractVector,
+  uhtF::AbstractVector,
+  uhttF::AbstractVector,
+  dutt_u::Real,
+  ode_cache)
+  Uh,Uht,Uhtt, = ode_cache
+  uh = EvaluationFunction(Uh,uhF)
+  uht = EvaluationFunction(Uht,uhtF)
+  uhtt = EvaluationFunction(Uhtt,uhttF)
+  jacobian_t!(J,op.feop,t,uh,uht,uhtt,dutt_u,ode_cache)
+end
+
+function jacobian_and_jacobian_t!(
+  J::AbstractMatrix,
+  op::SecondOrderODEOpFromFEOp,
+  t::Real,
+  uhF::AbstractVector,
+  uhtF::AbstractVector,
+  uhttF::AbstractVector,
+  dut_u::Real,
+  dutt_u::Real,
+  ode_cache)
+  Uh,Uht,Uhtt, = ode_cache
+  uh = EvaluationFunction(Uh,uhF)
+  uht = EvaluationFunction(Uht,uhtF)
+  uhtt = EvaluationFunction(Uhtt,uhttF)
+  jacobian_and_jacobian_t!(J,op.feop,t,uh,uht,uhtt,dut_u,dutt_u,ode_cache)
+end
```

Comment:
I would probably put it into another file. Again, I think that with the changes I propose above there is no need to replicate all the code from N=1 to N=2.

Reply:
No need for new functions. The functions have been generalized to N-th order ODEs